components/ThemeSwitcher/ThemeSwitcher.tsx

```diff
@@ -4,8 +4,11 @@ import useOnClickOutside from "../../customHooks/useOnClickOutside"
 
 import styles from "./ThemeSwitcher.module.css"
 
+type Theme = "light" | "dark" | "system" | undefined
```

Comment:
`system` is used to know when the user really selects the `Device settings`. `System` would have been ignored if the `device settings` isn't on the UI.

![Screenshot (141)](https://user-images.githubusercontent.com/70102539/197390568-d4276c21-31dc-4ac6-91bb-cb35c719f7d0.png)

The undefined is provided because the theme won't be available on the initial load, since there is no access to both `local storage` and `document`.